sds_data_manager/lambda_code/SDSCode/indexer.py

```diff
@@ -20,6 +20,146 @@
 s3 = boto3.client("s3")
 
 
+def s3_event_handler(event, engine):
+    """Handler function for S3 events.
+
+    Parameters
+    ----------
+    event : dict
+        The JSON formatted document with the data required for the
+        lambda function to process
+    engine : sqlalchemy.engine.Engine
+        Engine object to interact with the database.
+
+    Returns
+    -------
+    dict
+        HTTP response
+    """
+    # Retrieve the Object name
+    s3_filepath = event["detail"]["object"]["key"]
+
+    filename = os.path.basename(s3_filepath)
+    # TODO: add checks for SPICE or other
+    # data types
+    science_file = ScienceFile(filename)
+
+    if science_file.is_valid is False:
+        return {"statusCode": 400, "body": science_file.error_message}
+
+    # setup a dictionary of metadata parameters to unpack in the
+    # file catalog table
+    metadata_params = science_file.get_file_metadata_params()
+    metadata_params["file_path"] = s3_filepath
+
+    if metadata_params["data_level"] == "l0":
```

Comment:
invert this logic and emit the bad message right away and dedent the current if-block.
```suggestion
    if metadata_params["data_level"] != "l0":
        return {"statusCode": 400, "body": "Invalid data level"}
```